nucypher/policy/models.py

```diff
@@ -494,27 +503,39 @@ def __len__(self):
 
     @classmethod
     def construct_by_bob(cls, arrangement_id, capsules, ursula, bob):
-        receipt_bytes = b"wo:" + ursula.canonical_public_address  # TODO: represent the capsules as bytes and hash them as part of the receipt, ie  + keccak_digest(b"".join(capsules))  - See #137
+        capsules_bytes = [bytes(c) for c in capsules]
```

Comment:
yeah, that was my first try. I always tend to follow a functional approach (maps, reduces, filters, etc), but in this case,  if one uses `map`, then `msgpack.dumps` fails. Anyway, as much as I love functors, I think list comprehension reads cleaner in this case.

Reply:
`msgpack` wont consume a generator?  Or a genexp `(bytes(c) for c in capsules)`.  Not hugely important unless we can implement a longer chain of iterators, but, since this can be relatively high-throughput, it may be a worthwhile optimization.

Reply:
Meh, I think this wouldn't do much of a difference unless `capsules` involves some networking